kaldb/src/main/java/com/slack/kaldb/logstore/LogMessage.java

```diff
@@ -165,13 +167,13 @@ public boolean equals(Object o) {
       LOG.warn("id missing - equals comparison won't be accurate");
     }
     return timeSinceEpochMilli == that.timeSinceEpochMilli
-        && Objects.equal(getIndex(), that.getIndex())
-        && Objects.equal(getType(), that.getType())
-        && Objects.equal(id, that.id);
+        && Objects.equals(getIndex(), that.getIndex())
+        && Objects.equals(getType(), that.getType())
+        && Objects.equals(id, that.id);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hashCode(timeSinceEpochMilli, getIndex(), getMillisecondsSinceEpoch(), id);
+    return Objects.hash(timeSinceEpochMilli, getIndex(), getMillisecondsSinceEpoch(), id);
```

Comment:
Can we please make sure that the there a unit test testing this logic and that the test passes after this change,